app/models/mute.rb

```diff
@@ -0,0 +1,37 @@
+class Mute < ApplicationRecord
+  include MuteHelper
+
+  belongs_to :muter, class_name: "User"
+  belongs_to :muted, class_name: "User"
+
+  validates :muter, :muted, presence: true
+  validates :muted_id, uniqueness: { scope: :muter_id }
+
+  validate :check_self
+  validate :check_official, if: :muted
+  validate :check_mute_limit
+
+  after_create :update_cache
+  after_destroy :update_cache
+
+  def check_self
+    errors.add(:muted, :self) if muted == muter
+  end
+
+  def check_official
+    errors.add(:muted, :official) if muted.official
+  end
+
+  def check_mute_limit
+    errors.add(:muted, :limit) if muter.muted_users.length >= ArchiveConfig.MAX_MUTED_USERS
```

Comment:
`length` is not a good function to use here, as it'll force all of the muted users to be loaded. `count` is probably the right choice.